### PR TITLE
Tracker should be able to search just * and show all datasets

### PR DESCRIPTION
**Pull request by sushil-car**
https://issues.cask.co/browse/TRACKER-177

### CODE REVIEW

**Comment by sushil-car** [comment]
Added the search by * functionality in Tracker and changed the search API accordingly
Search API throwing below error when passed **sort=creation-time desc**  as parameter 

**Sort field must be 'name'. Found creation-time**

Please review

File: cdap-ui/app/tracker/controllers/results-ctrl.js [hunk 1]
```diff
@@ -137,6 +137,14 @@ class TrackerResultsController {
       scope: this.$scope
     };
 
+    if(params.query === '*') {
```
Comment (elfenheart) [inline cdap-ui/app/tracker/controllers/results-ctrl.js]: nit: space after `if`

File: cdap-ui/app/tracker/services/my-tracker-api.js [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 function myTrackerApi(myCdapUrl, $resource, myAuth, myHelpers, CDAP_UI_CONFIG) {
   var url = myCdapUrl.constructUrl,
-      searchPath = '/namespaces/:namespace/metadata/search?target=stream&target=dataset&target=view',
+      searchPath = '/namespaces/:namespace/metadata/search?limit=25&offset=0&target=dataset&target=stream',
```
Comment (elfenheart) [inline cdap-ui/app/tracker/services/my-tracker-api.js]: why was `target=view` removed?

File: cdap-ui/app/tracker/controllers/results-ctrl.js [hunk 3]
```diff
@@ -137,6 +137,14 @@ class TrackerResultsController {
...
+      params.sort = 'creation-time desc';
+      params.numCursors = 10;
+    } else {
+      params.query = params.query + '*';
+      params.target = 'app';
```
Comment (elfenheart) [inline cdap-ui/app/tracker/controllers/results-ctrl.js]: I don't think this is correct. Tracker should not search `app`

Comment (sushil-car) [reply cdap-ui/app/tracker/controllers/results-ctrl.js]: As provided on https://issues.cask.co/browse/TRACKER-177, it should be in params when we dont search with *
Could you please confirm

File: cdap-ui/app/tracker/services/my-tracker-api.js (hunk 2 above)
Comment (sushil-car) [reply cdap-ui/app/tracker/services/my-tracker-api.js]: As provided on https://issues.cask.co/browse/TRACKER-177, its not in params list
Could you please confirm

File: cdap-ui/app/tracker/controllers/results-ctrl.js (hunk 3 above)
Comment (ajainarayanan) [reply cdap-ui/app/tracker/controllers/results-ctrl.js]: Sushil: It was just an example. The targets represents what CDAP entity types the user is searching for. Please change as adviced by @elfenheart

File: cdap-ui/app/tracker/services/my-tracker-api.js (hunk 2 above)
Comment (ajainarayanan) [reply cdap-ui/app/tracker/services/my-tracker-api.js]: Same as the other comment

**Comment by elfenheart** [comment]
👍  LGTM

**Comment by ajainarayanan** [comment]
@sushiladokar Please squash the commits before merge. The general pattern can be found in other UI PRs. (_**[JIRA-####] - Short title of the bug being fixed**_)

**Comment by sushil-car** [comment]
@ajainarayanan 
Squashing the commits.

**Comment by russorat** [comment]
@sushiladokar why hasn't this been merged yet?

**Pull request merged**